pytensor/link/jax/dispatch/tensor_basic.py

```diff
@@ -51,9 +51,10 @@ def allocempty(*shape):
 
 
 @jax_funcify.register(Alloc)
-def jax_funcify_Alloc(op, **kwargs):
+def jax_funcify_Alloc(op, node, **kwargs):
     def alloc(x, *shape):
         res = jnp.broadcast_to(x, shape)
+        Alloc._check_runtime_broadcast(node, jnp.asarray(x), res.shape)
```

Comment:
When the actual check happens? every time the function runs?

Reply:
This is executed everytime JAX recompiles (if input shapes change). Otherwise it's not present in the jitted function whatsoever. This is what we already do for Elemwise and SpecifyShape for example.